### PR TITLE
[CRI] Return success if ImageNotFound in RemoveImage()

### PR DESCRIPTION
**Pull request by Crazykev**
Signed-off-by: Crazykev <crazykev@zju.edu.cn>

**What this PR does / why we need it**:

RemoveImage() operation should be idempotent, [ref](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/api/v1alpha1/runtime/api.proto#L89-L92)
 @feiskyer PTAL

**Which issue this PR fixes**

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/dockertools/kube_docker_client.go [hunk 1]
```diff
@@ -380,7 +381,14 @@ func (d *kubeDockerClient) RemoveImage(image string, opts dockertypes.ImageRemov
 	if ctxErr := contextError(ctx); ctxErr != nil {
 		return nil, ctxErr
 	}
-	return resp, err
+	if err != nil {
```
Comment (feiskyer) [inline pkg/kubelet/dockertools/kube_docker_client.go]: dockertools is about to remove in next release, so we don't need to change its codes.

Comment (Crazykev) [inline pkg/kubelet/dockertools/kube_docker_client.go]: @feiskyer Updated, I've moved these into dockershim

**Comment by feiskyer** [comment]
@k8s-bot ok to test

**Comment by feiskyer** [comment]
@k8s-bot gci gce e2e test this
@k8s-bot gce etcd3 e2e test this

**Comment by feiskyer** [comment]
@k8s-bot gci gce e2e test this
@k8s-bot verify test this

**Comment by feiskyer** [comment]
cc @yujuhong @Random-Liu PTAL.

**Comment by Crazykev** [comment]
moved to #45266

**Pull request closed without merging**